torchmultimodal/modules/encoders/clip_vit_encoder.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn, Tensor
+
+# Taken from original clip implementation https://github.com/openai/CLIP/blob/main/clip/model.py#L167
+# TODO: unify with the implementation in text encoder
+def quick_gelu(x: Tensor) -> Tensor:
+    return x * torch.sigmoid(1.702 * x)
+
+
+class CLIPViTEncoder(nn.Module):
+    """
+    Vision transformer encoder for CLIP.
+
+    Args:
+        embedding_dim (int): Embedding dimension for text and image encoders projections.
+        patch_size (int): The dimension of each patch
+        image_size(int): The size (width==height) of input image
+        width (int): Dimensionality of the encoder layers and the pooler layer
+        heads (int): Number of attention heads for each attention layer in the Transformer encoder
+        layers (int): Number of hidden layers in the Transformer encoder
+
+    Inputs:
+        x (Tensor): image tensor with dimensions B x C(3) x image_size x image_size
+    """
+
+    def __init__(
+        self,
+        embedding_dim: int,
+        patch_size: int,
+        image_size: int,
+        width: int,
+        heads: int,
+        layers: int,
+    ):
+        super().__init__()
+        self.conv = nn.Conv2d(
+            in_channels=3,
+            out_channels=width,
+            kernel_size=patch_size,
+            stride=patch_size,
+            bias=False,
+        )
+        self.image_size = image_size
+
+        scale = width**-0.5
+        self.cls_token_embedding = nn.Parameter(scale * torch.randn(width))
+        self.positional_embedding = nn.Parameter(
+            scale * torch.randn((image_size // patch_size) ** 2 + 1, width)
+        )
+        self.ln_pre = nn.LayerNorm(width)
```

Comment:
I notice in MMF CLIP they do an fp32 layernorm here. Do we wanna do something similar? (Not sure if ViSE is using mixed precision or not)

Reply:
yeah i saw that, was avoiding it till it was needed. but nvm, changed it now.